### PR TITLE
qemu_v8: Introducing MEMTAG option

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Introduces the option MEMTAG which enabled tagging of memory using the
ARM Memory Tagging Extension (MTE).

Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

**Review by jbech-linaro — COMMENTED** [review]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: qemu_v8.mk [hunk 1]
```diff
@@ -441,7 +456,7 @@ run-only:
 		-nographic \
 		-serial tcp:localhost:54320 -serial tcp:localhost:54321 \
 		-smp $(QEMU_SMP) \
-		-s -S -machine virt,secure=on,gic-version=$(QEMU_GIC_VERSION),virtualization=$(QEMU_VIRT) \
+		-s -S -machine virt,secure=on,mte=$(QEMU_MTE),gic-version=$(QEMU_GIC_VERSION),virtualization=$(QEMU_VIRT) \
```
Comment (jforissier) [inline qemu_v8.mk:459]: There is a similar line in `qemu-check.exp`.

**Comment by jenswi-linaro** [comment]
I've updated `qemu-check.exp` with MTE. When I try `make check MEMTAG=y -j8` it fails with:
```
Starting QEMU... done, guest is booted.
Running: xtest...
##################################################
##################################################
#############################################!!! assertion 'memtag_get_tag(addr) == memtag_read_tag(addr)' failed at lib/libutils/ext/include/memtag.h:214 in memtag_assert_tag()
make: *** [Makefile:483: check] Error 1
```

The assert is part of one of the test cases so I'm a bit surprised we error out here.

**Comment by jforissier** [comment]
> The assert is part of one of the test cases so I'm a bit surprised we error out here.

IIRC there is some output that the Expect script always considers as errors, "assertion" is probably one of them. We may want to relax that a bit.

**Comment by jforissier** [comment]
For commit "qemu-check.exp: trap only OP-TEE Core assert or panic":

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jforissier** [comment]
For commit "qemu_v8: Introducing MEMTAG option":

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
Squashed and tags applied

**Comment by jenswi-linaro** [comment]
Good to merge?

**Comment by jforissier** [comment]
> Good to merge?

Yes, thanks!

**Pull request merged**